### PR TITLE
 fix(infrastructure): types generated directory problem

### PR DESCRIPTION
**Pull request by TroyTae**
- fix lint/compile error
 - set "declarationDir" in tsconfig
 - add "types" property in dialog
 - add validate code that to check existence of definition file

### CODE REVIEW

File: scripts/release/verify-pkg-main.js [hunk 1]
```diff
@@ -31,37 +31,66 @@ const isValidCwd = fs.existsSync('packages');
...
     invalidMains++;
   }
+  if (isTypesInvalid) {
+    invalidTypes++;
+  }
 });
-
-if (invalidMains > 0) {
-  console.error(`${invalidMains} incorrect main property values found; please fix.`);
+if (invalidMains > 0 || invalidTypes > 0) {
```
Comment (moog16) [inline scripts/release/verify-pkg-main.js]: @TroyTae I updated this - let me know if you disagree with anything.

File: test/unit/menu/MenuList.test.tsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import {assert} from 'chai';
 import * as td from 'testdouble';
 import {shallow, mount} from 'enzyme';
 import {MenuList} from '../../../packages/menu/index';
-import List from '../../../packages/list/index';
+import List from '../../../packages/list';
```
Comment (moog16) [inline test/unit/menu/MenuList.test.tsx]: I don't think we should be doing this...I see the innerRef error, but this is indicating to us that there is an error with our types

File: scripts/release/verify-pkg-main.js (hunk 1 above)
Comment (TroyTae) [reply scripts/release/verify-pkg-main.js]: I agree to declare invalidTypes.
But we can remove isTypesInvalid and use isInvalid..!

Comment (moog16) [reply scripts/release/verify-pkg-main.js]: There are 2 different messages though.

File: test/unit/menu/MenuList.test.tsx (hunk 2 above)
Comment (TroyTae) [reply test/unit/menu/MenuList.test.tsx]: Ah I see, then let's fix this another PR!

File: scripts/release/verify-pkg-main.js (hunk 1 above)
Comment (TroyTae) [reply scripts/release/verify-pkg-main.js]: I got it now!

**Pull request merged**